org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xerces/RelaxNGErrorHandler.java

```diff
@@ -0,0 +1,58 @@
+package org.eclipse.lemminx.extensions.xerces;
+
+import org.apache.xerces.impl.XMLErrorReporter;
+import org.apache.xerces.xni.XNIException;
+import org.apache.xerces.xni.parser.XMLErrorHandler;
+import org.apache.xerces.xni.parser.XMLParseException;
+
+public class RelaxNGErrorHandler implements XMLErrorHandler {
+
+	private XMLErrorReporter report;
+
+	public RelaxNGErrorHandler(XMLErrorReporter report) {
+		this.report = report;
+	}
+
+	@Override
+	public void warning(String domain, String key, XMLParseException exception) throws XNIException {
+		report.reportError(domain, key, null, XMLErrorReporter.SEVERITY_WARNING, exception);
+	}
+
+	@Override
+	public void fatalError(String domain, String key, XMLParseException exception) throws XNIException {
+		report.reportError(domain, key, null, XMLErrorReporter.SEVERITY_FATAL_ERROR, exception);
+	}
+
+	@Override
+	public void error(String domain, String key, XMLParseException exception) throws XNIException {
+		if (domain.isEmpty() && key.isEmpty()) {

```

Comment:
Is it possible to do something like https://github.com/eclipse/lemminx/pull/1294/files?w=1#diff-06f18df1cea1fad3788617953e31457f65cb9f4a3093c08046fea88eb0b21784R320 here to get an "error code" for the message?